lib/LTP/utils.pm

```diff
@@ -181,38 +188,58 @@ sub check_kernel_taint {
         'ACPI table overridden by user',
         'Kernel issued warning',
         'Staging driver was loaded',
-        undef,    # platform firmware bug workaround, ignore
-        undef,    # out-of-tree module, ignore
+        'Workaround for platform firmware bug',
+        'Out of tree module was loaded',
         'Unsigned module was loaded',
         'Soft lockup occurred',
-        undef,    # livepatch, ignore
-        'Auxiliary taint',
-        undef,    # kernel built with struct randomization, ignore
-        'In-kernel test has been run'
+        'Kernel was live patched',
+        'Externally supported module was loaded or auxiliary taint',
+        'Kernel was built with struct randomization',
+        'In-kernel test has been run',
+        # currently unused taint flags
+        undef, undef, undef, undef, undef, undef, undef, undef, undef, undef,
+        undef, undef,
+        'Unsupported module was loaded'
     );
     my $flag = 1;
-    my @taint;
+    my $taint_undef = 0;
+    my $taint_mask = parse_int(get_var('LTP_TAINT_EXPECTED', 0x80019801));
```

Comment:
Maybe document (a comment or a constant or in the commit message or in variables.md) which taints are these (sure one can calculate them).

Reply:
Added comment with the list of taint flags in the default mask.